### PR TITLE
np.dot implementation for Phi and Gamma Tensors 

### PR DESCRIPTION
**Pull request by ivyclare**
## Description
Dot implementation for phi and gamma tensors

## Affected Dependencies
None 

## How has this been tested?
Yes, tests has been added for both

## Checklist
- [ x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ x] My changes are covered by tests

### CODE REVIEW

**Review by IshanMi — CHANGES_REQUESTED** [review]
Well done @ivyclare! Really terrific job for a first PR 👍 🎉 

I caught a few errors, and left comments in-line. Your PR also gave me a few ideas for things we could improve!
One thing that could be improved in the future would be to also test scenarios where a given operation is expected to fail. Additionally, since `dot` does different things depending on the shapes of the arrays involved, it might also be worthwhile to test the dot product of arrays of different shapes.

You did a great job and are learning fast. Well done, and keep up the good work!

File: packages/syft/tests/syft/core/tensor/adp/gamma_tensor_test.py [hunk 1]
```diff
@@ -128,3 +128,30 @@ def deduct_epsilon_for_user(*args: Any, **kwargs: Any) -> bool:
     print(results)
     print(results, results.dtype)
     print(ledger_store.kv_store)
+
+
+def test_pos(
```
[the remark this answers is not in the document]
Comment (ivyclare) [reply packages/syft/tests/syft/core/tensor/adp/gamma_tensor_test.py:133]: Hhmm looks like all gamma tests test the sum operation as a way to ensure that the gamma tensor was properly implemented.

**Review by IshanMi — CHANGES_REQUESTED** [review]
Just one or two small changes (and a few print statements to remove) and we can merge your first PR! Great work @ivyclare 👍

**Comment by IshanMi** [comment]
Closing this PR since we implemented dot during model training

**Pull request closed without merging**